packages/sdk/src/lib/dataProtectorSharing/rentProtectedData.ts

```diff
@@ -47,32 +51,65 @@ export const rentProtectedData = async ({
   let userAddress = await iexec.wallet.getAddress();
   userAddress = userAddress.toLowerCase();
 
-  const sharingContract = await getSharingContract(
-    iexec,
-    sharingContractAddress
-  );
+  const [sharingContract, pocoContract] = await Promise.all([
+    getSharingContract(iexec, sharingContractAddress),
+    getPocoContract(iexec),
+  ]);
 
   //---------- Smart Contract Call ----------
-  const protectedDataDetails = await getProtectedDataDetails({
-    sharingContract,
-    protectedData: vProtectedData,
-    userAddress,
-  });
+  const [protectedDataDetails, accountDetails] = await Promise.all([
+    getProtectedDataDetails({
+      sharingContract,
+      protectedData: vProtectedData,
+      userAddress,
+    }),
+    getAccountDetails({
+      pocoContract,
+      userAddress,
+      sharingContractAddress,
+    }),
+  ]);
 
   //---------- Pre flight check ----------
   onlyProtectedDataCurrentlyForRent(protectedDataDetails);
   onlyValidRentingParams(
     { price, duration },
     protectedDataDetails.rentingParams
   );
+  onlyAccountWithMinimumBalance({
+    accountDetails,
+    minimumBalance: vPrice,
+  });
 
   try {
     const { txOptions } = await iexec.config.resolveContractsClient();
-    const tx = await sharingContract.rentProtectedData(
-      vProtectedData,
-      { price: vPrice, duration: vDuration },
-      txOptions
-    );
+
+    let tx: ContractTransactionResponse;
+    const rentProtectedDataCallParams: [
+      AddressLike,
+      {
+        price: BigNumberish;
+        duration: BigNumberish;
+      }
```

Comment:
No need to specify the type here as it will be checked via typechain type-checking. This may increase the code's footprint from my perspective.

Same for other files